File: src/services/EmployeeService.js
import axios from "axios";

const BASE_URL = "http://localhost:8080/employees";

class EmployeeService {

    getAllEmployees() {
        return axios.get(BASE_URL);
    }

    getEmployeeById(employeeId) {
        return axios.get(`${BASE_URL}/${employeeId}`);
    }

    addEmployee(employee) {
        return axios.post(BASE_URL, employee);
    }

    updateEmployee(employeeId, employee) {
        return axios.put(
            `${BASE_URL}/${employeeId}`,
            employee
        );
    }

    deleteEmployee(employeeId) {
        return axios.delete(
            `${BASE_URL}/${employeeId}`
        );
    }
}

export default new EmployeeService();

File: src/main/java/com/employee/management/service/EmployeeService.java
package com.employee.management.service;



import java.util.List;
import java.util.Optional;

import com.employee.management.entities.Employee;

public interface EmployeeService {

    // Add Employee
    Employee addEmployee(Employee employee);

    // Get All Employees
    List<Employee> getAllEmployees();

    // Get Employee By ID
    Optional<Employee> getEmployeeById(String employeeId);

    // Update Employee
    Employee updateEmployee(String employeeId, Employee employee);

    // Delete Employee
    void deleteEmployee(String employeeId);

}